presto-native-execution/src/test/java/com/facebook/presto/nativeworker/ContainerQueryRunnerUtils.java

```diff
@@ -0,0 +1,372 @@
+
+            entryPointScript.toFile().setExecutable(true);
+            return entryPointScript;
+        }
+        catch (IOException e) {
+            fail("Presto CLI failed to create entrypoint.sh and was exited with error message: " + e.getMessage());
+        }
+        return null;
+    }
+
+    static Path createNodeProperties(String nodeId)
+    {
+        try {
+            Path tempNodeProperties = Files.createTempFile("node", ".properties");
+
+            String nodeProperties =
+                    "node.environment=testing\n" +
+                            "node.location=testing-location"
+                            + System.lineSeparator();
+
+            Files.write(tempNodeProperties, nodeProperties.getBytes(), StandardOpenOption.WRITE);
+
+            String nodeIdEntry = "node.id=" + nodeId + System.lineSeparator();
+            Files.write(tempNodeProperties, nodeIdEntry.getBytes(), StandardOpenOption.APPEND);
+            return tempNodeProperties;
+        }
+        catch (IOException e) {
+            fail("Presto CLI failed to create node.properties and was exited with error message: " + e.getMessage());
+        }
+        return null;
+    }
+
+    public static void printFileContents(Path filePath)
+            throws IOException
+    {
+        // Read all lines from the file
+        List<String> lines = Files.readAllLines(filePath);
+
+        // Print each line
+        for (String line : lines) {
+            System.out.println(line);
+        }
+        System.out.println("\n");
+    }
+
+    public static MaterializedResult toMaterializedResult(String csvData)
+    {
+        List<MaterializedRow> rows = new ArrayList<>();
+        List<Type> columnTypes = new ArrayList<>();
+
+        // Split the CSV data into lines
+        String[] lines = csvData.split("\n");
+
+        // Parse all rows and collect them
+        List<String[]> allRows = parseCsvLines(lines);
+
+        // Infer column types based on the maximum columns found
+        int maxColumns = allRows.stream().mapToInt(row -> row.length).max().orElse(0);
+        for (int i = 0; i < maxColumns; i++) {
+            final int columnIndex = i; // Make index effectively final
+            columnTypes.add(inferType(allRows.stream().map(row -> columnIndex < row.length ? row[columnIndex] : "").collect(Collectors.toList())));
+        }
+
+        // Convert all rows to MaterializedRow
+        for (String[] columns : allRows) {
+            List<Object> values = new ArrayList<>();
+            for (int i = 0; i < columnTypes.size(); i++) {
+                values.add(i < columns.length ? convertToType(columns[i], columnTypes.get(i)) : null);
+            }
+            rows.add(new MaterializedRow(5, values));
+        }
+
+        // Create and return the MaterializedResult
+        return new MaterializedResult(rows, columnTypes);
+    }
+
+    private static List<String[]> parseCsvLines(String[] lines)
```

Comment:
CSV parsing is notoriously error prone. Consider using a library like commons CSV instead: https://commons.apache.org/proper/commons-csv/

Reply:
Updated the code to using the library.